rasa/core/training/loading.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os
+from typing import Text, Optional, Dict, List, Any
+
+import rasa.utils.io as io_utils
+from rasa.core.domain import Domain
+from rasa.core.interpreter import NaturalLanguageInterpreter, RegexInterpreter
+from rasa.core.training.structures import StoryStep
+
+logger = logging.getLogger(__name__)
+
+
+def _get_reader_class(filename: Text) -> Optional[Any]:
+
+    from rasa.core.training.story_reader import yaml_story_reader, markdown_story_reader
+
+    module = None
+    if filename.endswith(".md"):
+        module = markdown_story_reader.MarkdownStoryReader
+    elif filename.endswith(".yml"):
+        module = yaml_story_reader.YAMLStoryReader
+
+    return module
+
+
+async def load_data_from_folder(
+    folder: Text,
+    domain: Domain,
+    interpreter: NaturalLanguageInterpreter = RegexInterpreter(),
+    template_variables: Optional[Dict] = None,
+    use_e2e: bool = False,
+    exclusion_percentage: Optional[int] = None,
+):
+    if not os.path.exists(folder):
+        raise ValueError(f"File '{folder}' does not exist.")
+
+    story_steps = await load_data_from_files(
+        io_utils.list_files(folder),
+        domain,
+        interpreter,
+        template_variables,
+        use_e2e,
+        exclusion_percentage,
+    )
+    return story_steps
+
+
+async def load_data_from_files(
+    story_files: List[Text],
+    domain: Domain,
+    interpreter: NaturalLanguageInterpreter = RegexInterpreter(),
+    template_variables: Optional[Dict] = None,
+    use_e2e: bool = False,
+    exclusion_percentage: Optional[int] = None,
+) -> List[StoryStep]:
+    story_steps = []
+
+    for story_file in story_files:
+
+        reader_class = _get_reader_class(story_file)
+        if not reader_class:
+            logger.warning(
+                f"Couldn't find a story reader for {story_file}, it will be skipped."
+            )
+            continue
+
+        reader = reader_class(
```

Comment:
@federicotdn That's somewhat related to the empty constructor. What do you think about it? I honestly don't know 🙈

Reply:
What is currently implemented works with or without the implicitly-defined initializer though. But maybe we can slightly change this so that when doing `if filename.endswith(".md") ...` etc. we already return an instance of the `Reader` subclass. We do this already in e.g. `broker._create_from_endpoint_config`.

Reply:
I like Fede's idea to simply create an instance inside the `_get_reader_class()`.
I think I was a bit in hurry and simply copied this from the NLU 😄